lib/Core/NvmHeuristics.h

```diff
@@ -0,0 +1,606 @@
+#ifndef __NVM_HEURISTICS_H__
```

Comment:
Definitely eventually want a big header comment here describing how the heuristic works to make it readable. I gave up on reading this code and trust that it's doing what you want.

Reply:
I will do this at some point, I pwomise.